model_signing/serializing/dfs_test.py

```diff
@@ -0,0 +1,282 @@
+
+    def test_directory_model_with_only_known_file(self, sample_model_file):
+        file_hasher = file.FileHasher("unused", memory.SHA256())
+        serializer = dfs.DFSSerializer(file_hasher, memory.SHA256)
+
+        model = sample_model_file.parent
+        manifest = serializer.serialize(model)
+
+        expected = (
+            "aa856f565699473579c8d7009bfad8c421e1643b810f0a28d47b9ce1f0b98ccc"
+        )
+        assert manifest.digest.digest_hex == expected
+
+        digest = memory.SHA256(_KNOWN_MODEL_TEXT).compute()
+        assert manifest.digest.digest_hex != digest.digest_hex
+
+    def test_known_folder(self, sample_model_folder):
+        file_hasher = file.FileHasher("unused", memory.SHA256())
+        serializer = dfs.DFSSerializer(file_hasher, memory.SHA256)
+        manifest = serializer.serialize(sample_model_folder)
+        expected = (
+            "516de24dd65c9749bbde333545cb997c645e21c510107fa5c06428e0df84099b"
+        )
+        assert manifest.digest.digest_hex == expected
+
+    def test_folder_model_hash_is_same_if_model_is_moved(
+        self, sample_model_folder
+    ):
+        file_hasher = file.FileHasher("unused", memory.SHA256())
+        serializer = dfs.DFSSerializer(file_hasher, memory.SHA256)
+        manifest = serializer.serialize(sample_model_folder)
+
+        new_name = sample_model_folder.with_name("new-root")
+        new_model = sample_model_folder.rename(new_name)
+        new_manifest = serializer.serialize(new_model)
+
+        assert manifest == new_manifest
+
+    def test_empty_file(self, empty_model_file):
+        file_hasher = file.FileHasher("unused", memory.SHA256())
+        serializer = dfs.DFSSerializer(file_hasher, memory.SHA256)
+        manifest = serializer.serialize(empty_model_file)
+        expected = (
+            "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+        )
+        assert manifest.digest.digest_hex == expected
+
+    def test_directory_model_with_only_empty_file(self, empty_model_file):
+        file_hasher = file.FileHasher("unused", memory.SHA256())
+        serializer = dfs.DFSSerializer(file_hasher, memory.SHA256)
+        manifest = serializer.serialize(empty_model_file)
+        model = empty_model_file.parent
+        manifest = serializer.serialize(model)
+        expected = (
+            "ca671f6b24ce1b08677759ed050a30eb86a28c18abfa2308c7da9e581a8f7917"
+        )
+        assert manifest.digest.digest_hex == expected
+
+    def test_empty_folder(self, empty_model_folder):
+        file_hasher = file.FileHasher("unused", memory.SHA256())
+        serializer = dfs.DFSSerializer(file_hasher, memory.SHA256)
+        manifest = serializer.serialize(empty_model_folder)
+        expected = (
+            "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+        )
+        assert manifest.digest.digest_hex == expected
+
+    def test_empty_folder_hashes_the_same_as_empty_file(
```

Comment:
maybe add an additional test that changes a non-empty file to a dir with the same name

Reply:
Good idea. Added.